### ISSUE TITLE
rust-format-buffer breaks markers in buffer

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust-mode#260]
For example

1. Open this file *a.rs*:
   ```rust
   // some function
   fn foo() {        }
   // some other function
   fn bar() {}
   ```
2. `M-x grep` and search for `fn` in *a.rs*.
3. `M-x next-error` (or `M-g n`), which moves the mark to line 2.
4. `M-x rust-format-buffer`, which removes the excess spaces between the braces in line 2.
5. `M-x next-error` moves the mark to the beginning of the buffer instead of line 4.

As a workaround, I can insert step 4.5: save buffer, and reopen it with `M-x find-alternate-file` (or `C-x C-v`), after which the broken markers for the buffer are lost and the next `M-x next-error` will move the mark to line 4. This is not perfect, as any edits that change the line numbers of matches will now cause an offset, but it is much better than having all errors sending me to the beginning of the buffer.

So if there is a way to invalidate all the markers in the buffer after a `M-x rust-format-buffer` which changes the buffer, that would basically solve this issue.

**Comment by Dushistov** [comment]
May be relate to #256

**Comment by nikomatsakis** [comment]
I've noticed this too, it's annoying =)

**Comment by mookid** [comment]
@tspiteri I can't reproduce any more, can we close the issue?

**Comment by tspiteri** [comment]
Yes, it should be fixed on emacs >= 26.2. For me that is enough: I have no opinion on whether the issue should be kept open because of emacs 25 and 26.1.
